### ISSUE TITLE
Convert icons to svg

### ISSUE THREAD

**Issue by weisJ** [MyLibreLab/MyLibreLab#27]
Currently icons are either `16x16` or `24x24`. For high resolution screens this isn’t an appropriate resolution anymore. Replacing the current icons with svg version solves this issue and will allow us to move further with updating the look and feel (more context as soon as I get the the laf integrated).

All currently used icons are located at [resources/Bilder](https://github.com/MyLibreLab/MyLibreLab/tree/master/src/main/resources/Bilder). Some images are available multiple times with different resolutions. They will only be replaced once of course.

Also some of those icons need to be renamed to have a more descriptive name. For some I don’t even know what they depict or used for so that needs to be investigated beforehand.

Generally I think we should aim for a more modern/flat design for the icons, similar to those of IntelliJ.
Note: The IntelliJ icons are available as svg versions to download and are compatible with our license, so we should be fine to use them. A list of all icons can be found [here](https://jetbrains.design/intellij/resources/icons_list/)
___
This could be marked as good first issue.

**Comment by shubhamgoyall** [comment]
Hey. Can I work on this??

**Comment by weisJ** [comment]
@shubhamgoyall yes absolutely. Help is highly appreciated.

**Comment by shubhamgoyall** [comment]
if there are 2 files.....save16.png and save24.png. They will be replaced by only one file...save.svg right??

**Comment by weisJ** [comment]
@shubhamgoyall Yes those files will be replaced by one file.

**Comment by markolo25** [comment]
@shubhamgoyall  you still working on this?

**Comment by shubhamgoyall** [comment]
> @shubhamgoyall  you still working on this? 

no..i unassigned myself

**Comment by markolo25** [comment]
ok i'll take over it

**Comment by marko-radosavljevic** [comment]
@markolo25 are you still working on this? If you are, what's the status?

**Comment by markolo25** [comment]
@marko-radosavljevic will work on it friday night, work's been pretty busy, and haven't had time, feel free to unassign so someone else can pick it up before friday.

**Comment by markolo25** [comment]
@marko-radosavljevic  it seems @weisJ has already made progress on this should we assign it to him instead.

**Comment by weisJ** [comment]
@markolo25 My PR is only to allow incremental work on the icons. Instead of merging some icons into dev they will first be merged into #60. Then if all icons are complete those can be merged together into the main branch.
The only icon I have added is the project logo :D

**Comment by markolo25** [comment]
thanks for the clarification, so workflow wise, i should fork #60 then create PR against #60? then once we're done we create a PR for #60 against dev?

**Comment by marko-radosavljevic** [comment]
I think #60 source branch should be on this repo, not in the fork. This is more confusing I believe, so lets wait for @weisJ to confirm if this is intentional or not ^^

**Comment by weisJ** [comment]
I didn't realize I pushed the branch to the wrong repo. I made a new PR (#64) with the branch on this repo.

**Comment by alehel** [comment]
Some icons have more than 1 variant. For instance edit-redo and redo. Is there any point in having different icons in cases like this? Could we just use the same redo icon for both situations? [edit] another example is application_osx_terminal and utilities-terminal.

**Comment by weisJ** [comment]
No there is no point in having different icons. One variant is enough.

**Comment by alehel** [comment]
How is GitHubs support for squashing a PR upon merge? Let's say we have a total of 10 PRs which merge icons into the main _convert to svg branch_. Could we then easily squash all these commits into one single commit against dev?

**Comment by weisJ** [comment]
Yes github allows for squash merge.

**Comment by alehel** [comment]
Tiny problem with the the underline icon. Intellij doesn't seem to have one. They have icons for bold and italic. Anyone know where we can find one for underline which would fit in with the ones intellij provides for bold and italic?

**Comment by weisJ** [comment]
I think there is an icon an A in intelliJ. We could shrink that one and add a plain rectangular line under it.
i.e. AllIcons.Actions.MatchCaseHovered maybe using just the small a as it gives us more play with the icon positioning.

See guidelines for icon design [here](https://jetbrains.org/intellij/sdk/docs/basics/plugin_structure/plugin_icon_file.html)

**Comment by alehel** [comment]
I always thought a U with a line under it was synonymous with underline, but this perhaps varies based on region?

**Comment by weisJ** [comment]
You are right. There is an "u" in CidrLangIcons.Icons.FileTypes.Cuh alternatively If this doesn't look good a small u could be fabricated out of one of the various small "m" in the icons. There might also be a small "u" somewhere. I only skimmed over the list.

**Comment by alehel** [comment]
I'll look into it. Hopefully I can finish another 20 tomorrow evening.

**Comment by weisJ** [comment]
I have updated the list of missing icons in the PR (#64).

**Comment by alehel** [comment]
Do we want a new netbeans logo? It's mentioned in the list. However, I assume it was mentioned because Netbeans was used to develop the program originally. As we are moving away form using Netbeans for the gui, do we still need this icon?

I also see the list mention an icon called icon. This is the O from the original logo. What do we want to replace this with?

**Comment by weisJ** [comment]
> Do we want a new netbeans logo? It's mentioned in the list. However, I assume it was mentioned because Netbeans was used to develop the program originally. As we are moving away form using Netbeans for the gui, do we still need this icon?

No we won’t need won’t a NetBeans logo.

> I also see the list mention an icon called icon. This is the O from the original logo. What do we want to replace this with?

We should try to create a simplified version of our new logo which is suitable to be used as a smaller logo. But for now the new logo can be used instead.

**Comment by alehel** [comment]
Anyone know what the zip icon is used for? I have a feeling not many people even know what a zip disc is any more 😂.

**Comment by weisJ** [comment]
The icons isn't actually used anywhere. At least I couldn't find any reference to it in the code.

**Comment by alehel** [comment]
Well that certainly makes it easier.

**Comment by alehel** [comment]
We have three flags that need icons (US, DE and ES). Been looking around for good candidates. I figured it was important we took them from a place that has a complete collection so that we can easily get new flags for other countries and still have them look consistent. I found this repo which has SVGs of flags for 249 countries. They originate from Wikimedia Commons. https://github.com/emcrisostomo/flags

The three flags are here:
https://github.com/emcrisostomo/flags/blob/master/svg/DE.svg
https://github.com/emcrisostomo/flags/blob/master/svg/ES.svg
https://github.com/emcrisostomo/flags/blob/master/svg/US.svg

Does this look like a good source for country icons?

**Comment by weisJ** [comment]
Looks like a good source to me. It would be good though to have all icons be the same aspect ration (e.g. through a square size). Otherwise we would need to hardcode the sizes for each icon separately when loading them.

**Comment by alehel** [comment]
Ah right, that didn't occur to me. I'll look into this.

**Comment by alehel** [comment]
What do we do about the ajax-loader which is basically an animated spinner?

**Comment by weisJ** [comment]
There is a [loading indicator](https://github.com/weisJ/darklaf/blob/master/core/src/main/java/com/github/weisj/darklaf/components/loading/LoadingIndicator.java) in darklaf which offers an animated spinner.

**Comment by alehel** [comment]
Ah right, we can skip that one then. 

Been looking into flags. Would the best option be to simply "squish" the flags to make all sides equal length? Some flags might look strange, but I can't really think of any other reliable way of handling it.

**Comment by weisJ** [comment]
Why not fix the flags width (or height whatever is larger. For most flags that should be the width) and have the rest transparent. The flags themselves don't need to be the same size only the icons including the negative space.

**Comment by alehel** [comment]
Ah. it appears I completely misunderstood what you meant 🤣. Yes, I can fix that no problem.

As for LGPL 3 and Universidad Catolica logos I did in fact find SVGs of these. Obviously these aren't 1:1 aspect ratio, but this may not be a problem for these two logos as I assume they won't be used as icons in the program, but rather in the about screen?

https://upload.wikimedia.org/wikipedia/commons/5/55/University_Cat%C3%B3lica_of_Colombia_logo.svg
https://upload.wikimedia.org/wikipedia/commons/3/3b/LGPLv3_Logo.svg

**Comment by weisJ** [comment]
> Ah. it appears I completely misunderstood what you meant 🤣. Yes, I can fix that no problem.
> 

Glad we could sort that out then 😃.

> As for LGPL 3 and Universidad Catolica logos I did in fact find SVGs of these. Obviously these aren't 1:1 aspect ratio, but this may not be a problem for these two logos as I assume they won't be used as icons in the program, but rather in the about screen?
> [...]

It's not necessary to have each icon in an 1:1 aspect ration. I just didn't want to hardcode the size for a group of possibly many icons. For one off icons it's perfectly reasonable to have specific sizes. 👍🏻
